src/test/daml/Daml/Finance/Instrument/Option/Test/Util.daml

```diff
@@ -71,45 +71,45 @@ originateDividendOption depository issuer label description observers lastEventT
     createReference cid depository issuer observers
 
 -- | Verify the payments of an instrument after an election (excluding settlement)
-electAndVerifyPaymentEffects : [Party] -> Date -> Decimal -> InstrumentKey -> Bool -> Party -> Party
+electAndVerifyPaymentEffects : Date -> Decimal -> InstrumentKey -> Bool -> Party -> Party -> Party
   -> Text -> [InstrumentQuantity] -> [InstrumentQuantity]
   -> Script (Optional InstrumentKey, ContractId Effect.I)
-electAndVerifyPaymentEffects readAs today amount instrument isLong issuer custodian
+electAndVerifyPaymentEffects today amount instrument electorIsOwner issuer custodian elector
   electedTag expectedConsumedQuantities expectedProducedQuantities = do
 
-    -- Create election offers to allow holders to create elections
+    -- Create election factory to allow holders to create elections
     electionFactoryCid <- submit issuer do
       toInterfaceContractId @Election.F <$> createCmd Election.Factory with
         provider = issuer
-        observers = M.fromList [("Holders", fromList [custodian] )]
+        observers = M.fromList [("Observers", fromList [elector, issuer] )]
 
-    -- Apply election to generate new instrument version + effects
+    -- Create a lifecycle rule
     lifecycleRuleCid <- toInterfaceContractId <$> submit custodian do
```

Comment:
Who is signing the instrument in your scenario? 

I could imagine the rule being provided by `issuer` rather than `custodian`

Reply:
Currently, this function is used in `Option.Test.EuropeanPhysical` and `Equity.Test.RightsIssue`, where there is a custodian as depository:

```

  itmLongCallInstrument <- originateEuropeanPhysicalOption custodian issuer "Long ITM Call"
...
  rightsInstrument <- originateEuropeanPhysicalOption custodian issuer "Long Call" "Option" pp now

```

Would it maybe make sense to change that?

Reply:
Makes sense, I have changed this now.